addons/hw_drivers/iot_handlers/drivers/L10nKeEDISerialDriver.py

```diff
@@ -149,8 +159,22 @@ def send(self, msgs):
                 request = struct.pack('B%ds2sB' % (len(core_message)), STX, core_message, self.generate_checksum(core_message), ETX)
                 time.sleep(self._protocol.commandDelay)
                 self._connection.write(request)
-                time.sleep(self._protocol.measureDelay)
-                response = self._connection.read_all()
+                # If we know the expected output size, we can set the read
+                # buffer to match the size of the output.
+                output_size = COMMAND_OUTPUT_SIZE.get(msg[0])
+                if output_size:
+                    try:
+                        response = self._connection.read(output_size)
+                    except serial.serialutil.SerialTimeoutException:
+                        _logger.exception('Timeout error while reading response to command %s', msg)
+                        self.data['status'] = "device timeout error"
```

Comment:
so the user will just see devie timeout error?

Reply:
yep

Reply:
Why not put a nicer message?

Reply:
```suggestion
                        self.data['status'] = "Device timeout error"
```